BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/XFintoPUEO_Symmetric.py

```diff
@@ -0,0 +1,133 @@
+    #pulling data from the uan file                                                                                         
+    data = np.genfromtxt(uanname, unpack=True, skip_header=18).tolist()
+
+    return (data[col]) #returns Gain(Theta) column THIS WILL NEED TO BE CHANGED FOR WHEN WE HAVE WORKING XF OUTPUTS (maybe make column input variable for function?)                                                                               
+## Make a function to create a list for each of the columns (V and H gains)
+## This function reads in the gain at each phi for the desired thetas
+def getGains(indiv):
+	vpolGains = []
+	hpolGains = []
+
+	for freq in range(numFreq):
+		vpolGains.append(readFile(indiv, freq+1, 2))
+		hpolGains.append(readFile(indiv, freq+1, 3))
+
+	return vpolGains, hpolGains
+
+## This function writes the data to a file
+def writeGains(data, freqList, fileName):
+	with open(fileName, "a") as f:
+		for i in range(numFreq):
+			f.write(str(freqList[i]) + " " + str(data[i]) + "\n")
+	return 0
+
+
+## Here's the path to output to
+fileHeader = g.WorkingDir / 'Test_Outputs' 
+## Loop over the number of individuals
+for indiv in range(1, g.NPOP+1):
+		## Let's call the functions
+		## Start by getting the gain data
+		vpolData, hpolData = getGains(indiv)
+		## We're treating vv and hh as the same and vh and hv as the same
+		## So for the _0 files, it's pretty simple:
+		writeGains(np.transpose(vpolData).tolist()[0], freqList, fileHeader / f'vv_0_{g.gen}_{indiv}')
+		writeGains(np.transpose(vpolData).tolist()[0], freqList, fileHeader / f'hh_0_{g.gen}_{indiv}')
+		writeGains(np.transpose(hpolData).tolist()[0], freqList, fileHeader / f'vh_0_{g.gen}_{indiv}')
+		writeGains(np.transpose(hpolData).tolist()[0], freqList, fileHeader / f'hv_0_{g.gen}_{indiv}')
+		## Now we need to be more careful for the az and el files
+		## We want the angles 5, 10, 20, 30, 45, 90
+		## For the azimuth, this is just counting through the phi at 0 theta
+		## We just pick out the 2nd, 3rd, 5th, 7th, 10th, and 19th sublists
+		## So we call the writeGains function for each of those lists
+		indices = [1, 2, 4, 6, 9, 18]
+		for i in indices:
+			writeGains(np.transpose(vpolData).tolist()[i], freqList, fileHeader / f'vv_az_{g.gen}_{indiv}')
+			writeGains(np.transpose(hpolData).tolist()[i], freqList, fileHeader / f'hh_az_{g.gen}_{indiv}')
+		## For the elevation, we need to count through theta at 0 phi
+		## These are trickier, because theta increments only after all of the phi increments
+		## So if 0,0 (theta, phi) is in the 0th row, and 0,360 is in the 72nd row
+		## then 5,0 is in the 73rd row, and we increment by adding 73 for each theta step
+		indices = [73, 146, 292, 438, 657, 1314]
+		for i in indices:
+			writeGains(np.transpose(vpolData).tolist()[i], freqList, fileHeader / f'vv_el_{g.gen}_{indiv}')
+			writeGains(np.transpose(hpolData).tolist()[i], freqList, fileHeader / f'hh_el_{g.gen}_{indiv}')
```

Comment:
```suggestion
        ## Now we need to be more careful for the az and el files
        ## We want the angles 5, 10, 20, 30, 45, 90
        ## For the azimuth, this is just counting through the phi at 0 theta
        ## We just pick out the 2nd, 3rd, 5th, 7th, 10th, and 19th sublists
        ## So we call the writeGains function for each of those lists
        indices = [1, 2, 4, 6, 9, 18]
        for i in indices:
            writeGains(np.transpose(vpolData).tolist()[i], freqList, fileHeader / f'vv_az_{g.gen}_{indiv}')
            writeGains(np.transpose(hpolData).tolist()[i], freqList, fileHeader / f'hh_az_{g.gen}_{indiv}')
        ## For the elevation, we need to count through theta at 0 phi
        ## These are trickier, because theta increments only after all of the phi increments
        ## So if 0,0 (theta, phi) is in the 0th row, and 0,360 is in the 72nd row
        ## then 5,0 is in the 73rd row, and we increment by adding 73 for each theta step
        indices = [73, 146, 292, 438, 657, 1314]
        for i in indices:
            writeGains(np.transpose(vpolData).tolist()[i], freqList, fileHeader / f'vv_el_{g.gen}_{indiv}')
            writeGains(np.transpose(hpolData).tolist()[i], freqList, fileHeader / f'hh_el_{g.gen}_{indiv}')
```